shared/router-v2/router.shared.js

```diff
@@ -67,49 +70,45 @@ export const oldActionToNewActions = (action: any, navigation: any) => {
     case RouteTreeGen.switchRouteDef: {
       // used to tell if its the login one or app one. this will all change when we deprecate the old routing
       const routeName = action.payload.routeDef.defaultSelected === 'tabs.loginTab' ? 'loggedOut' : 'loggedIn'
-      const switchStack = NavigationActions.navigate({params: undefined, routeName})
 
       // You're logged out
       if (routeName === 'loggedOut') {
-        return [switchStack]
+        return [NavigationActions.navigate({params: undefined, routeName: 'loggedOut'})]
       }
 
       // When we restore state we want the following stacks
-      // [People, TheLastTabYouWereOn, MaybeAConversationIfTheLastTabYouWereOnIsChat]
-      let sa = [StackActions.push({params: undefined, routeName: 'tabs.peopleTab'})]
+      let sa = [NavigationActions.navigate({params: undefined, routeName: 'loggedIn'})]
 
       if (action.payload.path) {
         const p = action.payload.path.last
           ? action.payload.path.last()
           : action.payload.path[action.payload.path.length - 1]
 
-        // a chat, we want people/inbox/chat
+        // a chat, we want inbox/chat
         if (p === 'chatConversation') {
-          sa.push(StackActions.push({params: undefined, routeName: 'tabs.chatTab'}))
+          sa.push(NavigationActions.navigate({params: undefined, routeName: 'tabs.chatTab'}))
           sa.push(StackActions.push({params: undefined, routeName: 'chatConversation'}))
-        } else if (p !== 'tabs.peopleTab') {
-          sa.push(StackActions.push({params: undefined, routeName: p}))
+        } else {
+          sa.push(NavigationActions.navigate({params: undefined, routeName: p}))
         }
       }
 
       // validate sa
-      if (!sa.every(a => routes[a.routeName])) {
+      if (
+        !sa.every(a => a.routeName === 'loggedIn' || routes[a.routeName] || mobileTabs.includes(a.routeName))
+      ) {
         logger.error('Invalid route found, bailing on push', sa)
         sa = []
       }
 
-      // switch the switch and do a reset of the stack
-      // MUST pass undefined as key in reset else it won't work correctly
-      return sa.length
-        ? [switchStack, StackActions.reset({actions: sa, index: sa.length - 1, key: undefined})]
-        : [switchStack]
+      return sa
     }
     case RouteTreeGen.clearModals: {
       const numModals = getNumModals(navigation)
       return numModals ? [StackActions.pop({n: numModals})] : []
     }
     case RouteTreeGen.navigateUp:
-      return [StackActions.pop()]
+      return [NavigationActions.back()]
     case RouteTreeGen.navUpToScreen: {
       const fullPath = Constants._getFullRouteForNavigator(navigation.state)
```

Comment:
Can't you simplify this action with navactions.back?

Reply:
i don't understand